src/main/java/com/polishuchenko/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -49,4 +50,13 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(Book.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find book with id= " + id, e);
+        }
```

Comment:
I think its redundant. If you do not get some book you will throw some exception in service